WcaOnRails/app/assets/javascripts/registrations.js

```diff
@@ -68,3 +68,12 @@ onPage('registrations#add, registrations#do_add', function() {
     $('#events input[type="checkbox"]').prop('checked', true);
   });
 });
+
+onPage('registrations#create', function() {
+  // Hide the hint when the user selects an event
+  // or selects all events
+  $('.associated-events input[type="checkbox"], .select-all-events').click(function() {
+    // opacity:0 rather than display:none to avoid DOM shifting
+    $('.associated-events .select-hint').css('opacity', '0');
```

Comment:
What about `visibility:hidden`?

Reply:
Ah is that prefered? I default to use opacity but I guess visibility is more readable.